contracts/test/PoolFDTLosses.t.sol

```diff
@@ -0,0 +1,314 @@
+
+        uint256 leeDepositAmt;
+        (lexDepositAmt, leeDepositAmt) = getSplitDepositAmounts(lexDepositAmt, totalDepositAmt - leoDepositAmt - lizDepositAmt);
+
+        // LPs is putting up the funds into the Pool.
+        noAssert_mintFundsAndDepositIntoPool(leo, pool,  leoDepositAmt, leoDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(liz, pool,  lizDepositAmt, lizDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(lex, pool2, lexDepositAmt, lexDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(lee, pool2, leeDepositAmt, leeDepositAmt);
+
+        // Fund the loan by both pool delegates.
+        pat.fundLoan(address(pool),  address(custom_loan), address(dlFactory), usdc.balanceOf(pool.liquidityLocker()));
+        pam.fundLoan(address(pool2), address(custom_loan), address(dlFactory), usdc.balanceOf(pool2.liquidityLocker()));
+
+        // Drawdown of the loan
+        drawdown(custom_loan, bob, totalDepositAmt);
+
+        /// Time warp to make a payment.
+        hevm.warp(custom_loan.nextPaymentDue());
+
+        // Make partial payment of loan
+        doPartialLoanPayment(custom_loan, bob);
+
+        {
+            // Claim the funds.
+            uint256[7] memory claimInfo1 = pat.claim(address(pool),  address(custom_loan), address(dlFactory));
+            uint256[7] memory claimInfo2 = pam.claim(address(pool2), address(custom_loan), address(dlFactory));
+
+            // withdraw interest by the lps.
+            leo.withdrawFunds(address(pool));
+            liz.withdrawFunds(address(pool));
+            lex.withdrawFunds(address(pool2));
+            lee.withdrawFunds(address(pool2));
+
+            // Verify the interest.
+            withinDiff(
+                getClaimedInterestNetOfFees(claimInfo1[1], pool.delegateFee(), pool.stakingFee()),
+                usdc.balanceOf(address(leo)) + usdc.balanceOf(address(liz)),
+                1
+            );
+            
+            withinDiff(
+                getClaimedInterestNetOfFees(claimInfo2[1], pool2.delegateFee(), pool2.stakingFee()),
+                usdc.balanceOf(address(lex)) + usdc.balanceOf(address(lee)),
+                1
+            );
+        
+
+            // Time warp to default
+            hevm.warp(custom_loan.nextPaymentDue() + globals.defaultGracePeriod() + 1);
+
+            // Setting slippage high enough to let it through the liquidations.
+            gov.setMaxSwapSlippage(7000);
+
+            // At least one of the pool delegates should be able to trigger a default
+            assertTrue(
+                pat.try_triggerDefault(address(pool), address(custom_loan), address(dlFactory)) ||
+                pam.try_triggerDefault(address(pool2), address(custom_loan), address(dlFactory)),
+                "Should be able to trigger Loan default"
+            );
```

Comment:
This is a cool new way to use the `try` functions

Reply:
Short circuits ftw.